### DISCUSSION TITLE
Colored Logs in Airflow UI

### DISCUSSION THREAD

**Discussion by piyush-daga** [apache/airflow#19125] [Q&A] [answered]
We run DBT using Airflow and the rich coloring of DBT allows for a simplified view of what failed, passed and was skipped.

Upon researching, I found that Airflow only supports colored logging when the terminal is TTY.

1. Can we get Airflow to have colored logs in the Airflow Logs UI?
2. Is there a reason why we chose to not enable/support the same?
3. What are the common workarounds that the community uses to view colored logs if any?

Thanks

**Comment by potiuk** [comment, marked as answer]
The reason is because usually when there is no TTY, the logs intermingled with color escape sequence will mess the logs.

If you want you can override logging in any way you want via https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html#advanced-configuration - Airlfow uses standard Python logging and you can take a look on how airflow logging is configured and override it with your own formatters. There are plenty of guidelines on how to do it (it's standard Python) and you can browse airflow source code to take some inspiration. just look for CustomTTYColoredFormatter which Airflow uses and local_settings where it is configured.
